### ISSUE TITLE
AS3 fails to start up due to socket hang-up error 

### ISSUE THREAD

**Issue by andreykashcheev** [F5Networks/f5-appsvcs-extension#450]
### Environment
 * Application Services Version: 3.27.0
 * BIG-IP Version:  15.1.0.4 - 0.0.6

### Summary
While installing AS3 extension, the socket hang up error occurs when LX Extension Framework attempts to start/run AS3. Here is error message from /var/log/restnoded/restnoded.log:

```
Wed, 28 Apr 2021 19:34:00 GMT - severe: [appsvcs] {"message":"Failed to complete startup. Error: socket hang up\n    at createHangUpError (_http_client.js:331:15)\n    at Socket.socketOnEnd (_http_client.js:423:23)\n    at emitNone (events.js:111:20)\n    at Socket.emit (events.js:208:7)\n    at endReadableNT (_stream_readable.js:1064:12)\n    at _combinedTickCallback (internal/process/next_tick.js:138:11)\n    at process._tickCallback (internal/process/next_tick.js:180:9)","level":"error"}
```
This error causes AS3 extension never to start up and running; attempt to hit AS3 /info endpoint results as 404 Not Found
Testing shows that this error surfaces only with AS3 and FAST extensions; no errors observed when only DO, TS and CFE get installed. 
Also, I validated that this issue is surfacing with older AS3 versions; in my testing, I used a build from April 2020 

It takes me about ~6 attempts (i,e, install, uninstall) before I was able to replicate the error.

### Steps To Reproduce
   1) Create a script to install several extensions (i.e. DO, AS3, TS, CFE and FAST) 
```
curl -sk -u admin: -d '{"operation": "INSTALL","packageFilePath": "/var/lib/cloud/icontrollx_installs/f5-declarative-onboarding-1.20.0-2.noarch.rpm"}' http://localhost:8100/mgmt/shared/iapp/package-management-tasks
curl -sk -u admin: -d '{"operation": "INSTALL","packageFilePath": "/var/lib/cloud/icontrollx_installs/f5-cloud-failover-1.8.0-0.noarch.rpm"}' http://localhost:8100/mgmt/shared/iapp/package-management-tasks
curl -sk -u admin: -d '{"operation": "INSTALL","packageFilePath": "/var/lib/cloud/icontrollx_installs/f5-appsvcs-3.19.0-4.noarch.rpm"}' http://localhost:8100/mgmt/shared/iapp/package-management-tasks
curl -sk -u admin: -d '{"operation": "INSTALL","packageFilePath": "/var/lib/cloud/icontrollx_installs/f5-telemetry-1.19.0-3.noarch.rpm"}' http://localhost:8100/mgmt/shared/iapp/package-management-tasks
curl -sk -u admin: -d '{"operation": "INSTALL","packageFilePath": "/var/lib/cloud/icontrollx_installs/f5-appsvcs-templates-1.8.1-1.noarch.rpm"}' http://localhost:8100/mgmt/shared/iapp/package-management-tasks
```
   2) When installation is done, validate AS3 is up and running: 
```
curl -s -u admin: http://localhost:8100/mgmt/shared/appsvcs/info | jq .
```
  3) In a case when 404 Not Found response received from /info endpoint, grep restnoded logs using the following command: 
  
  ```
  egrep '(Startup Failed|Failed to complete|failed to register)' /var/log/restnoded/restnoded.log
  ```
   

Here is script which I used for replicating this issue: 
https://ak-f5-cft.s3-us-west-2.amazonaws.com/validate_lx_extension_installation.sh 

My investigation/troubleshooting indicates that the following steps can be used for mitigating this issue:
  1) adding delay ~15 secs between installation calls
  2) restart restnoded using "bigstart restart restnoded" command

### Expected Behavior
AS3 is up and running after installation

### Actual Behavior
AS3 fails to start and /info endpoint return 404 Not Found even though AS3 is installed successfully

**Comment by dstokesf5** [comment]
Thank you for your feedback. This bug has been added to our internal product backlog as AUTOTOOL-2469.

**Comment by amolari** [comment]
@dstokesf5 same issue here with as3 3.28.0 - in the context of installing it within runtime-init (v1.2.1)

**Comment by dstokesf5** [comment]
We have added some mitigations to 3.29.0 that should allow AS3 to start up much more often in cases where BIG-IP is in a state that causes these socket hang ups.

**Comment by dstokesf5** [comment]
This issue has been resolved by the 3.29.0 release.
